### PR TITLE
fix(Scripts/Zones/Creature): Updating Torturer Lecraft

### PR DESCRIPTION
**Pull request by Knindzagxg**
## Changes Proposed:

This PR proposes changes to:
-  [x] Core (units, players, creatures, game systems).
-  [x] Scripts (bosses, spell scripts, creature scripts).

## Issues Addressed:

- After some time creature will bug itself and will spam Kidney Shot on player no matter what.

## SOURCE:

The changes have been validated through:
- [x] Live research (checked on live servers, e.g Classic WotLK, Retail, etc.)
- [x] Video evidence, knowledge databases or other public sources (e.g forums, Wowhead, etc.)

## Tests Performed:

This PR has been:
- [x] Tested in-game by the author.
- [x] Tested in-game by other community members/someone else other than the author/has been live on production servers.

## How to Test the Changes:

- [x] This pull request can be tested by following the reproduction steps provided in the linked issue

1. .quest add 12252 (Add Quest to yourself)
2. .go cre 102544 (Teleport to creature)
3. Use quest item and watch hes AI.

## Known Issues and TODO List:

- [ ]
- [ ]

## How to Test AzerothCore PRs
 
When a PR is ready to be tested, it will be marked as **[WAITING TO BE TESTED]**.

You can help by testing PRs and writing your feedback here on the PR's page on GitHub. Follow the instructions here:

http://www.azerothcore.org/wiki/How-to-test-a-PR

**REMEMBER**: when testing a PR that changes something **generic** (i.e. a part of code that handles more than one specific thing), the tester should not only check that the PR does its job (e.g. fixing spell XXX) but **especially** check that the PR does not cause any regression (i.e. introducing new bugs).

**For example**: if a PR fixes spell X by changing a part of code that handles spells X, Y, and Z, we should not only test X, but **we should test Y and Z as well**.

### CODE REVIEW

**Comment by pvp100com** [comment]
I tested this pull request, but I didn't see any difference. 
But I unexpectedly discovered the following issues. about GUID = 97213 and 103288



Fixed:

```
UPDATE `creature_addon` SET `bytes1`=7, `emote`=65 WHERE  `guid`=97213;
UPDATE `creature_addon` SET `bytes1`=7, `emote`=65 WHERE  `guid`=103288;
```

**Comment by elthehablo** [comment]
Could you restructure the script to use the more modern form of a ScriptedAI struct?

See: https://github.com/azerothcore/azerothcore-wotlk/blob/master/src/server/scripts/Outland/boss_doomwalker.cpp as an example

**Comment by Knindzagxg** [comment]
Its just a part of the script and whole zone, not a single one.

File: src/server/scripts/Northrend/zone_dragonblight.cpp [hunk 1]
```diff
@@ -2236,11 +2241,11 @@ class npc_torturer_lecraft : public CreatureScript
                 {
                     case EVENT_HEMORRHAGE:
                         DoCastVictim(SPELL_HEMORRHAGE);
-                        _events.ScheduleEvent(EVENT_HEMORRHAGE, 12s, 168s);
+                        _events.ScheduleEvent(EVENT_HEMORRHAGE, urand(12000, 18000));
```
Comment (Winfidonarleyan) [inline src/server/scripts/Northrend/zone_dragonblight.cpp:2244]: Plase use `std::chrono` format

Comment (Knindzagxg) [reply src/server/scripts/Northrend/zone_dragonblight.cpp:2244]: Will sort in a day or two, cheers.

**Comment by sudlud** [comment]
what is the state of this PR? can you please address the review comments

File: src/server/scripts/Northrend/zone_dragonblight.cpp [hunk 2]
```diff
@@ -2252,6 +2257,8 @@ class npc_torturer_lecraft : public CreatureScript
         EventMap _events;
         uint8    _textCounter;
         ObjectGuid   _playerGUID;
+        uint32   _lastHemorageTime;
```
Comment (Nyeriah) [inline src/server/scripts/Northrend/zone_dragonblight.cpp:2260]: neither of those variables are currently being used by your script

**Comment by Nyeriah** [comment]
The creature is just missing an _events.Reset() call either on Reset() or enter combat.
It just keeps scheduling a new set of events every time its pulled that keep on repeating, which in turn leads to this: 
image

**Pull request closed without merging**